controller/src/lib.rs

```diff
@@ -232,8 +308,13 @@ impl Controller {
             message,
             data: None,
         };
-        let _reply = self.rpc(request).await.unwrap();
-        todo!();
+        let reply = self.rpc(request).await?;
+        Ok(reply
+            .data
+            .unwrap()
+            .into_iter()
+            .map(|x| Status::from_bits(x).unwrap())
+            .collect())
```

Comment:
Use `MissingData` here?

e.g.
```suggestion
        let reply = self.rpc(request).await?;
        let data = reply.data.ok_or(MessageError::MissingData)?;
        Ok(data
            .into_iter()
            .map(|x| Status::from_bits(x).unwrap())
            .collect())
```

Reply:
I think we could, but `self.rpc` should already check that the length of `data` is exactly what we expect. That's done [here](https://github.com/oxidecomputer/transceiver-control/blob/939b4b2d89571ef1ec0bef25bc8fb2ec8eaa4fd0/controller/src/lib.rs#L729-L755). Does that suffice?

Reply:
Aha, that was far enough away that I forgot about it.  Maybe use something like `.expect("checked earlier")` as a reminder?

Reply:
(although per [parse-don't-validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/), this suggests we should have a different type for "message with known-correct `data` member" 😃)

Reply:
Yeah, that's fair! The `SpResponse` type really _is_ the type representing that. It just needs a fallible constructor or something, so that we can't build one that _should_ carry data but does not.